vignettes/advanced_data.Rmd

```diff
@@ -0,0 +1,211 @@
+
+Now we can look at what we estimated.
+
+```{r plague inf res}
+fit_inf
+```
+
+We can see that $R$ is inferred to be above one with probability 1.
+It is likely near 1, though.
+We see considerable uncertainty about $k$, as we did in the borealpox example.
+
+### Treat it as censored
+
+We can alternately declare that all we know is that chain was _at least_ as big as we saw.
+That is, it didn't go extinct under its own power at size 5009, but it might have at 5010, or 7500, or something else.
+This is accomplished using the `ccensor_geq` argument, which specifies the smallest possible size that _each chain_ (allowing for as many censored observations as needed), with `NA` meaning no censoring is applied.
+(This is significantly slower, so for demonstration purposes only, we run fewer and shorter chains here.)
+
+```{r treat as censored, results=FALSE}
+censoring_sizes <- c(rep(NA, 3), 5009, rep(NA, 15))
+fit_cen <- fit_nbbp_homogenous_bayes(
+  all_outbreaks = pneumonic_plague,
+  condition_geq = rep(2, length(pneumonic_plague)),
+  censor_geq = censoring_sizes,
+  seed = 42,
+  iter = 500,
+)
+```
+
+As always, first we inspect the MCMC.
+As stan has warned us, performance isn't amazing, but it is good enough for demonstration purposes.
+
+```{r plague censored mcmc}
+rstan::check_hmc_diagnostics(fit_cen)
+
+rstan::summary(fit_cen)$summary[, c("n_eff", "Rhat")]
+```
+
+Now we can look at what we estimated.
+
+```{r plague censored res}
+fit_cen
+```
+
+We can see that now there is some probability that $R < 1$, corresponding to the possibility that this chain was simply large, but finite.
+We again see considerable uncertainty about $k$.
+
+### Comparison
+Obviously, the two analyses disagree on whether $R$ can be below 1.
+When we take the chain of 5009 to be non-extinct, this probability is 0, while when we take that chain to be censored, it is `r round(mean(unlist(rstan::extract(fit_cen, "r_eff")) < 1), 2) * 100`%.
+Otherwise, though, estimates of $R$ are qualitatively similar.
+
+```{r}
+r_df <- data.frame(
+  r = c(
+    unlist(rstan::extract(fit_inf, "r_eff")),
+    unlist(rstan::extract(fit_cen, "r_eff"))
+  ),
+  large_chain = c(
+    rep("non-extinct", 10000),
+    rep("censored", 1000)
+  )
+)
+
+r_df |>
+  ggplot(aes(x = r, fill = large_chain, y = after_stat(density))) +
+  geom_histogram(position = "identity", alpha = 0.7) +
+  theme_minimal() +
+  xlab("R") +
+  geom_vline(xintercept = 1.0, lty = 2)
+```
+
+This suggests that, in some sense, the model "prefers" to consider the chain non-extinct even if we did not say it was such.
+We can compare, across the posterior on $(R, k)$, the probabilities of the chain being large (but finite) versus non-extinct.
+We can get these with the PMF (`dnbbp`) and CDF (`pnbbp`) functions in the package for all posterior samples.
+(Here we use 500 samples for illustrative purposes, but in general one should use the entire posterior.)
```

Comment:
The reader ought to be familiar with negative binomial branching processes as a prerequisite, but this may be a good spot to remind the reader that a chain with R < 1 must go extinct eventually, whereas R > 1 only means that a chain might not go extinct (but it still can).